### PR TITLE
Use `URL` global in `parseUri`

### PR DESCRIPTION
**Pull request by jokull**
## Reproduction

1. Start a new next.js app router project (e.g. bun create next-app)
2. Add `app/test/route.ts`
3. Add this 

```ts
import { createClient } from "@libsql/client/web";
import { NextResponse } from "next/server";

export const runtime = "edge";

export async function GET() {
  const statement = await createClient({
    url: "http://127.0.0.1:8080",
  }).execute("SELECT 1");
  return NextResponse.json(statement.toJSON());
}
```

Run `bun run build`.

### CODE REVIEW

File: packages/libsql-core/src/uri.ts [hunk 1]
```diff
@@ -39,87 +39,21 @@ export interface KeyValue {
...
+        authority: {
+            userinfo: {
+                username: url.username,
+                password: url.password,
+            },
+            host: url.hostname,
+            port: url.port ? parseInt(url.port) : undefined,
+        },
+        path: url.pathname,
+        query: { pairs: [...url.searchParams.values()].map(([key, value]) => ({ key, value })) },
```
[the remark this answers is not in the document]
Comment (jokull) [reply packages/libsql-core/src/uri.ts:54]: Yes on `.entries()` ... my bad 😵‍💫 

But isn't `url.searchParams` the same as `new URLSearchParams(url.search)`?

[the remark this answers is not in the document]
Comment (giovannibenussi) [reply packages/libsql-core/src/uri.ts]: You're right. For some reason when I tested it with url.searchParams I got an error but probably was related to other part of the code that I was working with. We can stick `url.searchParams` and just update the function call to `entries`!

**Comment by jokull** [comment]
Dug into some tests. I don't believe the global `URL` class will handle the following URL's found in tests without some special cases and RegEx detection:

1. `file://localhost` - and its variants, because `URL` seems to have some special handling around not wanting to interpret hostnames
2. `file:` with relative paths
3. An empty path in any URL will get overwritten as the root `/` path

I'm no longer convinced my proposal is viable. It will need much more work to become backwards compatible and I don't have time for that right now.

**Comment by giovannibenussi** [comment]
Thanks a lot for your help with this. This is really useful for us because we still need to fix the [issue with Next.js](https://github.com/tursodatabase/libsql-client-ts/pull/160) so your insights help us a lot!

I want to test out two different ideas. The first one is to look for an already built 3986 parser like [elysiumphase/node-uri](https://github.com/elysiumphase/node-uri) as a replacement and the second idea is to find a workaround for the regex.

**Comment by penberg** [comment]
Fixed by https://github.com/tursodatabase/libsql-client-ts/pull/172 so closing this PR.

**Pull request closed without merging**